File: go/space-age/space_age.go
package space

type Planet string

const EarthSeconds = 31557600

var planets = map[Planet]float64{
	"Mercury": 0.2408467 * EarthSeconds,
	"Venus":   0.6151972 * EarthSeconds,
	"Earth":   EarthSeconds,
	"Mars":    1.8808158 * EarthSeconds,
	"Jupiter": 11.862615 * EarthSeconds,
	"Saturn":  29.447498 * EarthSeconds,
	"Uranus":  84.016846 * EarthSeconds,
	"Neptune": 164.79132 * EarthSeconds,
}

func Age(seconds float64, planet Planet) float64 {
	if v, ok := planets[planet]; ok {
		return seconds / v
	}
	return 0.0
}

File: elixir/space-age/space_age.exs
defmodule SpaceAge do
  @type planet :: :mercury | :venus | :earth | :mars | :jupiter
                | :saturn | :uranus | :neptune

  @earth_second 31_557_600
  @planets %{
  	mercury: 0.2408467,
  	venus:   0.6151972,
  	earth:   1,
  	mars:    1.8808158,
  	jupiter: 11.862615,
  	saturn:  29.447498,
  	uranus:  84.016846,
  	neptune: 164.79132,
  }

  @doc """
  Return the number of years a person that has lived for 'seconds' seconds is
  aged on 'planet'.
  """
  @spec age_on(planet, pos_integer) :: float
  def age_on(planet, seconds) do
    case Map.fetch(@planets, planet) do
      {:ok, planet_second} -> seconds / (planet_second * @earth_second)
      :error -> 0
    end
  end
end
